v-next/hardhat-keystore/src/index.ts

```diff
@@ -0,0 +1,82 @@
+import type { HardhatPlugin } from "@ignored/hardhat-vnext/types/plugins";
+
+import "./internal/type-extensions.js";
+
+import { task } from "@ignored/hardhat-vnext/config";
+import { ArgumentType } from "@ignored/hardhat-vnext/types/arguments";
+import { isCi } from "@ignored/hardhat-vnext-utils/ci";
+
+import { PLUGIN_ID } from "./internal/constants.js";
+
+export const hardhatKeystorePlugin: HardhatPlugin = {
+  id: PLUGIN_ID,
+  hookHandlers: {
+    config: import.meta.resolve("./internal/hook-handlers/config.js"),
+    configurationVariables: import.meta.resolve(
+      "./internal/hook-handlers/configuration-variables.js",
+    ),
+  },
+  tasks: [
+    task("keystore", "Store your keys in a secure way")
+      .setAction(async () => {})
+      .build(),
+
+    task(
+      ["keystore", "set"],
+      "Sets a new value in the keystore associated with the specified key",
+    )
+      .addPositionalArgument({
+        name: "key",
+        type: ArgumentType.STRING,
+        description: "Specifies the key to set in the keystore",
+      })
+      .addFlag({
+        name: "force",
+        description: "Forces overwrite if the key already exists.",
+      })
+      .setAction(import.meta.resolve("./internal/tasks/set.js"))
+      .build(),
+
+    task(["keystore", "get"], "Get a value given a key")
+      .addPositionalArgument({
+        name: "key",
+        type: ArgumentType.STRING,
+        description: "Specifies the key to retrieve the value for",
+      })
+      .setAction(import.meta.resolve("./internal/tasks/get.js"))
+      .build(),
+
+    task(["keystore", "list"], "List all keys in the keystore")
+      .setAction(import.meta.resolve("./internal/tasks/list.js"))
+      .build(),
+
+    task(["keystore", "delete"], "Delete a key from the keystore")
+      .addPositionalArgument({
+        name: "key",
+        type: ArgumentType.STRING,
+        description: "Specifies the key to delete from the keystore",
+      })
+      .addFlag({
+        name: "force",
+        description:
+          "Forces to not throw an error if the key does not exist during deletion.",
+      })
+      .setAction(import.meta.resolve("./internal/tasks/delete.js"))
+      .build(),
+  ],
+};
+
+const continuousIntegrationKeystorePlugin: HardhatPlugin = {
+  id: PLUGIN_ID,
+  hookHandlers: {},
+  tasks: [],
+};
+
+// The keystore plugin should be disabled in CI environments.
+// We implement this by returning a Keystore plugin
+// with its functionality turned off (i.e. no hook handlers or tasks)
+const chosenPlugin: HardhatPlugin = isCi()
+  ? continuousIntegrationKeystorePlugin
+  : hardhatKeystorePlugin;
+
+export default chosenPlugin;
```

Comment:
Note this is how CI disabling is implemented for the keystore plugin. We subtly replace the plugin returned on an `isCi()` detection.

The tests are careful to import not the `default` export but the full plugin directly.

Reply:
I think a better alternative is to just have

```
if (isCi()) {
  return next(...);
}
```

in the `fetchValue` hook handler.

Why? Because the current approach changes the CI setup significantly (e.g. some tasks disappear) and it's harder to communicate and understand.

Instead, if we just don't fetch the values, that'd be conceptually simpler.

Reply:
Done with this commit: [`e7b5ff9` (#5654)](https://github.com/NomicFoundation/hardhat/pull/5654/commits/e7b5ff98e49a0234cb4a7a084d55e2df10b6914d)

Reply:
Though I ended up having to put in some `process.env.CI` toggling magic in the tests as the Hook fetch value integration tests run in our CI.